src/js/index.d.ts

```diff
@@ -19,11 +19,11 @@ declare class UACustomEvent {
    * Adds a property to the custom event.
    *
    * @param {string} name The property name.
-   * @param {string|number|boolean|string[]} value The property value.
+   * @param {string|number|boolean|string[]|Object} value The property value.
```

Comment:
in typescript we should be using the non-primitive `object` type instead of `Object`: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#general-types

Reply:
I also think we can probably afford to be more strict with the typesystem here and constrain the keys and values the object can contain.

Reply:
e.g.

```
type JsonValue = string | number | boolean | null | JsonMap | JsonArray;
interface JsonMap {  [key: string]: JsonValue; }
interface JsonArray extends Array<JsonValue> {}
```

Reply:
Done.